new_lineage_panel/src/components/Column.tsx

```diff
@@ -0,0 +1,111 @@
+import { FunctionComponent, ReactElement, useMemo } from "react";
+import BooleanIcon from "../assets/icons/boolean.svg?react";
+import TextIcon from "../assets/icons/text.svg?react";
+import NumberIcon from "../assets/icons/number.svg?react";
+import ObjectIcon from "../assets/icons/object.svg?react";
+import DatetimeIcon from "../assets/icons/datetime.svg?react";
+import GeospatialIcon from "../assets/icons/geospatial.svg?react";
+import ModelIcon from "../assets/icons/model.svg?react";
+import SeedIcon from "../assets/icons/seed.svg?react";
+import SourceIcon from "../assets/icons/source.svg?react";
+import ExposureIcon from "../assets/icons/exposure.svg?react";
+import AnalysisIcon from "../assets/icons/analysis.svg?react";
+import SnapshotIcon from "../assets/icons/snapshot.svg?react";
+import MetricsIcon from "../assets/icons/metrics.svg?react";
+import MacrosIcon from "../assets/icons/macros.svg?react";
+import styles from "../styles.module.scss";
+import { UncontrolledTooltip } from "reactstrap";
+
+export const ColumnDatatype: FunctionComponent<{
+  datatype: string;
+  color?: string;
+  size?: string;
+}> = ({ datatype, color, size = "1rem" }) => {
+  const [Icon, compColor] = useMemo(() => {
+    switch (datatype.toLowerCase()) {
+      case "integer":
+      case "float":
+      case "double precision":
+      case "double":
+      case "bigint":
+        return [NumberIcon, "#FF754C"];
+      case "bool":
+      case "boolean":
+        return [BooleanIcon, "#00A5DB"];
+      case "text":
+      case "character":
+      case "character varying":
+      case "varchar":
+        return [TextIcon, "#3F8CFF"];
+      case "geospatial":
+        return [GeospatialIcon, "#01CD8C"];
+      case "date":
+      case "timestamp":
+      case "timestamp with time zone":
+        return [DatetimeIcon, "#247EFE"];
+      default:
+        return [ObjectIcon, "#6A24FE"];
+    }
+  }, [datatype]);
+
+  return (
+    <div
+      style={{ color: color || compColor }}
+      className="d-flex align-items-center"
+    >
+      <Icon width={size} height={size} />
+    </div>
+  );
+};
+
+export const NodeTypeIcon: FunctionComponent<{ nodeType: string }> = ({
+  nodeType,
+}) => (
+  <div>
+    {nodeType === "seed" && <SeedIcon />}
```

Comment:
performance tip: this code will execute all nodeType checks for every render. You could either return icon on match or create a json object with nodetype and icon and just return in one line
first approach:
```
export const NodeTypeIcon: FunctionComponent<{ nodeType: string }> = ({
  nodeType,
}) => {
if (nodeType === 'seed'){
return <seedIcon />
}
...
...
}
```

second approach:
```
const IconMap = {
seed: <SeedIcon />,
...
...
}
export const NodeTypeIcon: FunctionComponent<{ nodeType: string }> = ({
  nodeType,
}) => (
  <div>
{IconMap[nodeType]}
</div>
```